src/createCors.ts

```diff
@@ -7,13 +7,13 @@ export type CorsOptions = {
   headers?: any
 }
 
-export interface CorsFns {
-  corsify(req: Request): Response;
-  preflight(r: IRequest): Response;
+export interface CorsHandlers {
+  corsify(response: Response): any
+  preflight(Request: IRequest): any
```

Comment:
Never use a direct `IRequest` anywhere but as a default for a generic type. It'd be best if you setup a linter for this, tbh. It's the source of 99% of your type bugs.